### ISSUE TITLE
statement.insert() does not work when using two prepared statements

### ISSUE THREAD

**Issue by vandenoever** [jgallagher/rusqlite#171]
Currently, Statement::insert checks the id of the last inserted row and reports an error if that id has not changed after an insert. However, this fais when inserting in two tables alternatively. The first insert in table A can create rowid = 1, then a second insert in table B can also give back rowid = 1. Both operations are successful, but rusqlite will report an issue.
```rust
        let prev_rowid = self.conn.last_insert_rowid();  // rowid from insert in table A
        let changes = try!(self.execute(params));
        let new_rowid = self.conn.last_insert_rowid();  // rowid from insert in table B
        match changes {
            1 if prev_rowid != new_rowid => Ok(new_rowid),
            1 if prev_rowid == new_rowid => Err(Error::StatementFailedToInsertRow),
            _ => Err(Error::StatementChangedRows(changes)),
        }
```

**Comment by jgallagher** [comment]
Ahh, yeah, you're right. This was intended to catch `UPDATE`s run through `insert`, but maybe we have to just not check for that.

**Comment by vandenoever** [comment]
I'm going to use a workaround for now by simply calling stmt.execute(params). The code is very easy to read.

**Comment by jgallagher** [comment]
Just published 0.7.3 which fixes this issue. Thanks for reporting it!

**Comment by vandenoever** [comment]
That's an elegant solution. Thanks!
